### PR TITLE
Remove referrer policy from document

### PR DESCRIPTION
**Pull request by shanehandley**
This change removes the `referrer_policy` from document, instead referring calls to get and set the referrer_policy to Document's policy container ([spec change](https://github.com/whatwg/html/pull/6677)).

There are also some fixes to how referrer policies are handled during processing of script modules and their dependent scripts ([spec change](https://github.com/whatwg/html/pull/9210)).

It only fixes one of the new failure raised in [a previous PR](https://github.com/servo/servo/pull/33977#issuecomment-2449034213), but it makes it much easier to debug what's going on with one referrer_policy floating around.

[Try](https://github.com/shanehandley/servo/actions/runs/11866519095)

---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] There are tests for these changes OR

### CODE REVIEW

File: components/script/dom/document.rs [hunk 1]
```diff
@@ -3848,12 +3838,13 @@ impl Document {
...
 
     pub fn set_referrer_policy(&self, policy: Option<ReferrerPolicy>) {
-        self.referrer_policy.set(policy);
+        self.policy_container
+            .borrow_mut()
+            .set_referrer_policy(policy);
     }
 
-    //TODO - default still at no-referrer
     pub fn get_referrer_policy(&self) -> Option<ReferrerPolicy> {
```
Comment (jdm) [inline components/script/dom/document.rs:3846]: If we always return Some here, let's remove the Option from the return type.

**Review by jdm — APPROVED** [review]
Very nice!

File: components/script/dom/document.rs (hunk 1 above)
Comment (shanehandley) [reply components/script/dom/document.rs]: 👍 updated in https://github.com/servo/servo/pull/34263/commits/3104746061f18756021e1ab177820e3d9c8f38fc

[Updated try](https://github.com/shanehandley/servo/actions/runs/11886854303)

**Comment by jdm** [comment]
Once the clippy error is fixed we can merge!

**Comment by shanehandley** [comment]
Sorry @jdm I ran another try after fixing the clippy issue, and spotted a failure: https://github.com/shanehandley/servo/actions/runs/11887894147/job/33121864282

This may be an intermittent caused by my changes, I will dig deeper after my day job tomorrow 👍

**Comment by shanehandley** [comment]
> Sorry @jdm I ran another try after fixing the clippy issue, and spotted a failure: https://github.com/shanehandley/servo/actions/runs/11887894147/job/33121864282
> 
> This may be an intermittent caused by my changes, I will dig deeper after my day job tomorrow 👍

I've not been able to reproduce this failure, either in GH runners or locally.

**Pull request merged**